frontend/src/app/aethel/aethel.component.ts

```diff
@@ -45,21 +49,41 @@ export class AethelComponent implements OnInit {
                 }
                 this.rows = this.addUniqueKeys(response.results);
             });
+
+        // If the route has a query parameter, we run a search with that query immediately.
+        this.route.queryParams
```

Comment:
I don't mean to keep being difficult about `rxjs`, but in this case the following seems much simpler to me:

```
let params = new URLSearchParams(location.search);
if (params.has('query')) {
    this.form.controls.aethelInput.setValue(params.get('query'));
    this.search();
}
```

Reply:
You're right! Nice one :D

Reply:
Actually, with the dynamic queryParams that trigger searches, we do need a reactive approach, so I have kept the observable in place for now.